presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectStreamReader.java

```diff
@@ -144,47 +149,69 @@ public Block readBlock(Type type)
             }
         }
 
+        offsetVector[0] = 0;
         long totalLength = 0;
         for (int i = 0; i < nextBatchSize; i++) {
+            offsetVector[i + 1] = offsetVector[i];
             if (!isNullVector[i]) {
                 totalLength += lengthVector[i];
+                offsetVector[i + 1] += lengthVector[i];
```

Comment:
Updates to `offsetVector` are completely overwritten in the next loop. Remove these unnecessary updates. After removing these lines, move `offsetVector[0] = 0;` to the beginning of the next loop.